### PR TITLE
Add support for Linux desktop entries

### PR DESCRIPTION
**Pull request by ata2001**

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -39,49 +42,106 @@ def init(self, settings, q):
...
+                            try:
+                                app, command = self.parse_desktop_entry(desktop_entry)
+                            except KeyError:
+                                continue
+
+                            # TODO: Accept list of arguments
+                            for pattern, info in [(r'(?<!%)%[fF]', '<file name>'),
+                                                  (r'(?<!%)%[uU]', '<URL>')]:
+                                command_info, n = subn(pattern, info, command, count=1)
+                                if n:
```
Comment (TheLastProject) [inline __init__.py]: Could you explain what this code does?

File: __init__.py [hunk 2]
```diff
@@ -105,19 +164,26 @@ def selection_made(self, selection):
...
+            args = selection[0]["args"]
 
             if not self.use_path and platform.system() == 'Darwin':
-                Popen(["open", "-a", "{}".format(command)])
-            else:
+                if args:
+                    command = ("open", "-a", shlex.quote(command), "--args") + shlex.split(args)
+                else:
+                    command = ("open", "-a", shlex.quote(command))
+                Popen(command)
```
Comment (TheLastProject) [inline __init__.py]: Shouldn't command be a list instead of a tuple?

**Comment by ata2001** [comment]
I made the requested changes in the last 3 commits.

**Comment by TheLastProject** [comment]
When launching the App Launcher module, I get an error:

```
While reading from '/usr/share/applications/mimeapps.list' [line 209]: option 'x-content/video-dvd' in section 'Default Applications' already exists
```

Traceback:
```
Traceback (most recent call last):
  File "/home/sylvia/Projects/Pext/Pext/pext/__main__.py", line 1520, in run
    threading.Thread.run(self)
  File "/usr/lib64/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/home/sylvia/.config/pext/modules/pext_module_launcher/__init__.py", line 48, in init
    self._get_entries()
  File "/home/sylvia/.config/pext/modules/pext_module_launcher/__init__.py", line 117, in _get_entries
    app, command = self.parse_desktop_entry(desktop_entry)
  File "/home/sylvia/.config/pext/modules/pext_module_launcher/__init__.py", line 53, in parse_desktop_entry
    parser.read(path)
  File "/usr/lib64/python3.7/configparser.py", line 696, in read
    self._read(fp, filename)
  File "/usr/lib64/python3.7/configparser.py", line 1091, in _read
    fpname, lineno)
configparser.DuplicateOptionError: While reading from '/usr/share/applications/mimeapps.list' [line 209]: option 'x-content/video-dvd' in section 'Default Applications' already exists
```

It's probably a bad .desktop entry, but I'd still like it to not fail in such a case.

**Comment by TheLastProject** [comment]
From a quick search, probably want to set "strict" to False.

**Comment by TheLastProject** [comment]
It seems to be a combination of 2 .desktop files:

fedora-release-notes.desktop
```
[Desktop Entry]
Name=Fedora Release Notes
Categories=System;Documentation;
GenericName=Release Notes
Type=Application
X-Desktop-File-Install-Version=0.23
Exec=xdg-open file:///usr/share/doc/fedora-release-notes/index.html
Icon=fedora-release-notes
Comment=Release Notes for Fedora 28.01
NotShowIn=GNOME;
```

fedora-release-notes.webapp.desktop:
```
[Desktop Entry]
Name=Fedora Release Notes
Categories=System;Documentation;
GenericName=Release Notes
Type=Application
X-Desktop-File-Install-Version=0.23
Exec=sh -c "PROFILEDIR=\"\$HOME/.config/epiphany/fedora-release-notes\";DOC="/usr/share/doc/fedora-release-notes/index.html"; [ -d \$PROFILEDIR ] || mkdir \$PROFILEDIR;epiphany --application-mode --profile=\$PROFILEDIR \$DOC || xdg-open \$DOC " 
Icon=fedora-release-notes
Comment=Release Notes for Fedora 28.01
OnlyShowIn=GNOME;
```

**Comment by ata2001** [comment]
thanks, yes, it seems like pext parsed it correctly. If you select it does the same thing happen as if you launched it with your file manager?

**Comment by TheLastProject** [comment]
fedora-release-notes.desktop opens the release notes
fedora-release-notes.webapp.desktop opens http://doc

Pext causes the desktop to throw an error: "Malformed URL"

**Comment by TheLastProject** [comment]
Okay, that was a recent messup in Pext itself. Pull request approved :)

**Pull request merged**